### PR TITLE
DRIVERS-2025 Include the "clusterTime" field in ChangeStreamDocument

### PR DESCRIPTION
**Pull request by abr-egn**

### CODE REVIEW

**Review by benjirewis — CHANGES_REQUESTED** [review]
Awesome! You'll also want to update the "last updated" date, bump the minor version of the spec, and add to the bottom of the changelog. You could wait until the changes for DRIVERS-1343 are merged to do an additional spec version bump + changelog entry. 

Also, is there any way we can add a spec test to ensure `ChangeStreamDocument`s will include the new `clusterTime` field?

**Comment by abr-egn** [comment]
Sorry for the long delay here!  I've updated this to include version/last modified/changelog updates, and added a spec test.  The test [passes](https://spruce.mongodb.com/version/62ffa4f3d1fe0759ef85d3de/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC) a test run against all configurations in the Rust driver (the few failures there are unrelated).

File: source/change-streams/tests/unified/change-streams-clusterTime.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      id: &database0 database0
+      client: *client0
+      databaseName: *database0
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: *collection0
+
+runOnRequirements:
+  - minServerVersion: "4.0.0"
```
Comment (benjirewis) [inline source/change-streams/tests/unified/change-streams-clusterTime.yml:17]: Why 4.0.0?

**Review by benjirewis — APPROVED** [review]
Thanks for the update!

File: source/change-streams/tests/unified/change-streams-clusterTime.yml (hunk 1 above)
Comment (abr-egn) [reply source/change-streams/tests/unified/change-streams-clusterTime.yml:17]: The field was new as of 4.0 (see the description of DRIVERS-471).

**Pull request merged**